frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPartition.jsx

```diff
@@ -86,11 +86,17 @@ function SortOrderOption({ value, onChange }) {
 }
 
 function FormattingOptions({ onEdit }) {
+  const settingsRef = useRef(null);
+
+  const handleOnEdit = () => {
```

Comment:
I think it might be possible to use a target from an event from onClick. This would help simplify the code quite a bit by being able to remove useRef and keep the code relatively the same.